### DISCUSSION TITLE
bad function call runtime error when using class `GeneralPhase`

### DISCUSSION THREAD

**Discussion by Noxical** [reaktoro/reaktoro#281] [Q&A]
### Description

Hi there,
I try to set the aggregatestate manually using the setAggregateState method. However program gives a RunTime error: bad function call?

```
from reaktoro import *

db = NasaDatabase("nasa-cea")
    
general = GeneralPhase("H2O(l)").setAggregateState(AggregateState.Liquid)

system = ChemicalSystem(db, general)
```

### What is your operating system?

Windows

**Comment by allanleal** [comment]
Hi @Noxical , the `GeneralPhase` is a very low-level construct in Reaktoro. The following works:

~~~python

general = GeneralPhase("H2O(l)")
general.setName("LiquidPhase")
general.setStateOfMatter(StateOfMatter.Liquid)
general.setAggregateState(AggregateState.Liquid)
general.setActivityModel(ActivityModelIdealSolution(StateOfMatter.Liquid))
general.setIdealActivityModel(ActivityModelIdealSolution(StateOfMatter.Liquid))

general = general.convert(db, ["H", "O"])

system = ChemicalSystem(db, PhaseList([general]))
~~~

which is equivalent to:

~~~python
liquid = LiquidPhase("H2O(l)")

system = ChemicalSystem(db, liquid)
~~~

Make sure you indeed need to work with `GeneralPhase`. If so, use the above example as a starting point.

**Reply by Noxical** [reply]
Thanks for the explanation. Probably will not use, but I was bit confused about the construct.
